### ISSUE TITLE
[libxml2] libxml2:x64-windows-static is not statically linked with libiconv

### ISSUE THREAD

**Issue by clemenswasser** [microsoft/vcpkg#16298]
****

**Is your feature request related to a problem? Please describe.**

When linking `libxml:x64-windows-static` it produces errors like these, because it is not statically linked with `libiconv`:
```
lld-link: error: undefined symbol: libiconv_open
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1691
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1692
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1694
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced 1 more times

lld-link: error: undefined symbol: libiconv
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2827
>>>               libxml2.lib(encoding.c.obj):($LN45)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2611
>>>               libxml2.lib(encoding.c.obj):($retry$111)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2684
>>>               libxml2.lib(encoding.c.obj):($retry$111)
>>> referenced 9 more times

lld-link: error: undefined symbol: libiconv_close
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1703
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1704
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2744
>>>               libxml2.lib(encoding.c.obj):($LN29)
>>> referenced 1 more times
```

**Proposed solution**

Statically link to `libiconv` when building `libxml2:x64-windows-static`

**Describe alternatives you've considered**

Could be resolved by #15390

**Additional context**

**Comment by JackBoosY** [comment]
@clemenswasser It looks like you are using a cross environment.
The libxml2 cmake configuration file contains its dependencies:
```cmake
# Create imported target LibXml2::LibXml2
add_library(LibXml2::LibXml2 STATIC IMPORTED)

set_target_properties(LibXml2::LibXml2 PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "LIBXML_STATIC"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/libxml2"
  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:ws2_32>;Iconv::Iconv;\$<LINK_ONLY:LibLZMA::LibLZMA>;\$<LINK_ONLY:Threads::Threads>;\$<LINK_ONLY:ZLIB::ZLIB>"
)
```

but if you use cross-compilation, please manually add the dependencies to the link list.

**Comment by clemenswasser** [comment]
I get this error because I manually specified the CMake variables `-DLIBXML2_INCLUDE_DIR='<VCPKG_ROOT>/installed/x64-windows-static/include' -DLIBXML2_LIBRARY='<VCPKG_ROOT>/installed/x64-windows-static/lib/libxml2.lib'`, due to libxml2's [vcpkg-cmake-wrapper.cmake](https://github.com/microsoft/vcpkg/blob/master/ports/libxml2/vcpkg-cmake-wrapper.cmake#L21) being broken:
```
CMake Error at <VCPKG_ROOT>/installed/x64-windows-static/share/libxml2/vcpkg-cmake-wrapper.cmake:21 (target_link_libraries):
  Cannot specify link libraries for target "LibXml2::LibXml2" which is not
  built by this project.
Call Stack (most recent call first):
  <VCPKG_ROOT>/scripts/buildsystems/vcpkg.cmake:811 (include)
  <MY_PROJECT>/CMakeLists.txt:203 (find_package)
```
This Issue will probably be fixed by #15390

**Comment by clemenswasser** [comment]
I found a fix 🥳
You can just set `CMAKE_PREFIX_PATH` to `<VCPKG_ROOT>/installed/x64-windows-static` (where `VCPKG_ROOT` is your vcpkg folder, you have to use forward slashes!).
CMake will then figure out where the include directories are and what libraries to link against.

**Comment by clemenswasser** [comment]
This was fixed by #19120
